### DISCUSSION TITLE
Interrupt capable pins

### DISCUSSION THREAD

**Discussion by KChegwin** [earlephilhower/arduino-pico#357] [Q&A] [answered]
I'm working on a project using a generic RP2040, I would like to know if all the GPIO pins are interrupt capable using this board manager. For context, I am attempting to define interrupts for use with the encoder.h library (https://github.com/PaulStoffregen/Encoder/tree/master/utility ). I suspect I have to list all interrupt capable pins in order to achieve this and so a list of these would be much appreciated.

regards,

**Comment by earlephilhower** [comment, marked as answer]
The core should support all IRQ enabled GPIOs as the RP2040 itself (i.e. all of the digital I/Os).  If not, let me know and we can see what's up.

**Reply by KChegwin** [reply]
Thanks for this, I'll do a bit more work with the library's implementation of interrupts and see what I can discover before getting back in touch

**Reply by NuclearPhoenixx** [reply]
Sorry to revive this ancient thread, but did you get it to work eventually? I just tried to do something similar and the library won't compile neither with or without interrupts. I know it's more of an "issue" with the library than with this core, but I might as well ask you if you tried to do the same thing.
